definitions.h

```diff
@@ -226,14 +226,54 @@ class Attributes
         // Obligatory empty constructor
         Attributes() {}
 
-        PIXEL color;
+        // Initialize with colors
+        Attributes(double red, double green, double blue) : r(red), g(green), b(blue) {}
+
+        // Member variables
+        PIXEL color; // DEPRECATED - use r,g,b instead
+        double r;
```

Comment:
I might not understand the project right, but I thought we were supposed to right the code in a way that either rgb or uv would be used, not both at the same time for any triangle. I could be wrong though. Code looks clean, nice job